src/Routing/RouteBuilder.php

```diff
@@ -747,28 +747,27 @@ protected static function parseDefaults($defaults)
         if (preg_match($regex, $defaults, $matches)) {
             unset($matches[0]);
             $matches = array_filter($matches, function ($value) {
-                return $value !== '' && $value !== '::';
+                return $value !== '::';
```

Comment:
Why removing empty string check here? That was on purpose afaik.

Reply:
That's what was removing $matches[2]. With it there, it could mess with the switch as it relies on the count of the array verse specifically looking to see if like $matches[4] is set.

Reply:
Plus, the switch is referencing specific indexes within the $marches array, and only returning specific indexes, so if $matches[11] existed because this was taken it out, ultimately it wouldn't matter (other then maybe garbage collection if $matches is huge for some reason).

Reply:
Also, I guess it should return an empty string instead of null? 
See failed tests:
https://travis-ci.org/cakephp/cakephp/jobs/375121707

Reply:
This is a filter method on the array, any empty value is being stripped, same as :: which is similar empty value.